app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt

```diff
@@ -230,6 +235,18 @@ class SettingsViewModel @Inject constructor(
         }
     }
 
+    fun startPollingSyncEnableState() {
+        syncStateMonitor.syncState()
+            .onEach {
+                viewState.value = currentViewState().copy(
+                    syncEnabled = when (it) {
+                        OFF -> false
+                        else -> true
+                    },
+                )
+            }.launchIn(viewModelScope)
+    }
```

Comment:
Why do you need this method to be called from the Activity.
It is not really polling anything and you can attach to the activity lifecycle through the VM

Reply:
good catch, moved to the VM